### PR TITLE
implemented event deletion

### LINKED ISSUE #27: Event deletion
Right now there's no way to delete an existing event:
- add a trash icon beside each event in a calendar cell

### PR DESCRIPTION
**Pull request by larrycoal**
## Description

This PR implement event deletion 

## Issue

This PR fixes #27 

## Blocker

Has permission issue that probably needs to be fixed on firebase dev console

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,10 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@emotion/react": "^11.8.2",
+    "@emotion/styled": "^11.8.1",
+    "@mui/icons-material": "^5.5.1",
+    "@mui/material": "^5.5.2",
```
Comment (ANDREYDEN) [inline package.json:9]: I believe only `@mui/icons-material` is used in this PR, are other packages needed?

**Review by ANDREYDEN — CHANGES_REQUESTED** [review]
Looks great! I believe the issue with insufficient permissions is due to some work that I have done on the Auth branch #31. I'll merge that in ASAP and get back to you when that is ready.

**Comment by ANDREYDEN** [comment]
@larrycoal, sorry this took so long, but I finally merged the authentication (#31) branch. You should be able to test your functionality with Firebase Emulators (`npm run emulate`). Your branch now also has some conflicts, so you might want to merge `main` into your branch.

File: package.json (hunk 1 above)
Comment (ANDREYDEN) [reply package.json:9]: nvm, now I see they are needed

File: src/pages/calendar/Calendar.jsx [hunk 2]
```diff
@@ -1,68 +1,96 @@
...
-import InputField from '../../components/inputField/InputField'
-import { addEvent, listenToEvents } from '../../services/firestoreService'
-import './Calendar.css'
-
-function renderEvent(event) {
+import { useEffect, useState } from "react";
+import Modal from "react-modal";
+import { Delete } from "@mui/icons-material";
+import CustomButton from "../../components/customButton/customButton";
+import InputField from "../../components/inputField/InputField";
```
Comment (ANDREYDEN) [inline src/pages/calendar/Calendar.jsx:5]: These imports throw errors when you run the app

File: src/pages/calendar/Calendar.jsx [hunk 3]
```diff
@@ -71,38 +99,37 @@ function Calendar() {
 
       <h2>{months[selectedMonth]}</h2>
 
-      <CustomButton onClick={openEventModal} type="button" buttonText='+' />
+      <CustomButton onClick={openEventModal} type="button" buttonText="+" />
```
Comment (ANDREYDEN) [inline src/pages/calendar/Calendar.jsx:102]: This button should only be shown when the user is an admin (current behavior on `master`)

File: src/pages/calendar/Calendar.jsx [hunk 4]
```diff
@@ -1,68 +1,96 @@
...
+import "./Calendar.css";
+
+function renderEvent(event, handleDelete) {
   return (
-    <div key={event.id} className='event'>
+    <div key={event.id} className="event">
       <h3>{event.title}</h3>
       <p>{event.duration}h</p>
+      <div className="deleteEvent">
+        <Delete
```
Comment (ANDREYDEN) [inline src/pages/calendar/Calendar.jsx:19]: This button should only be shown for admins.

**Review by ANDREYDEN — CHANGES_REQUESTED** [review]
Something weird happened when you merged `master` into your branch resulting in 34 file differences. While it seems like most of the changes are just duplicates from previous commits, there are some places you might need to be careful with. Those are mostly the documents you edited to achieve the 'delete' functionality

File: src/pages/calendar/Calendar.jsx (hunk 2 above)
Comment (larrycoal) [reply src/pages/calendar/Calendar.jsx:5]: While trying to resolve conflict, i messed up my commit history

Comment (ANDREYDEN) [reply src/pages/calendar/Calendar.jsx:5]: @larrycoal, do you mind recreating the PR with only your changes or trying to merge `master` in again?

Comment (larrycoal) [reply src/pages/calendar/Calendar.jsx:5]: will start from scratch and open new PR

**Pull request closed without merging**